### PR TITLE
Added Java transceiver interface comments

### PR DESCRIPTION
**Pull request by bentoi**
This PR adds comments for the `setReadyCallback` and `fd` methods on the `Transceiver` interface.

### CODE REVIEW

File: java/src/Ice/src/main/java/com/zeroc/IceInternal/Transceiver.java [hunk 1]
```diff
@@ -5,8 +5,24 @@
...
   java.nio.channels.SelectableChannel fd();
 
+  /**
+   * Sets the transceiver ready callback. This method is called by the thread pool to provide
+   * a callback object that the transceiver can use to notify the thread pool's selector when
+   * more data is ready to be read or written by this transceiver.
+   *
+   * @param callback The ready callback provided by the thread pool's selector.
+   * @remark  The transceiver implementation typically uses this callback when it buffers
+   * data read from the selectable channel of if it doesn't use a selectable channel.
```
Comment (bernardnormier) [inline java/src/Ice/src/main/java/com/zeroc/IceInternal/Transceiver.java:24]: channel of it: looks like a typo?

File: java/src/Ice/src/main/java/com/zeroc/IceInternal/Transceiver.java [hunk 2]
```diff
@@ -5,8 +5,24 @@
...
+   * write readiness.
+   *
+   * @return The selectable channel that will be registered with the thread pool's selector
+   * or null if the transceiver doesn't use a selectable channel.
+   */
   java.nio.channels.SelectableChannel fd();
 
+  /**
+   * Sets the transceiver ready callback. This method is called by the thread pool to provide
+   * a callback object that the transceiver can use to notify the thread pool's selector when
```
Comment (bernardnormier) [inline java/src/Ice/src/main/java/com/zeroc/IceInternal/Transceiver.java:19]: "can use" should be uses?

I assume a correct implementation of Transceiver does not ignore this ReadyCallback. If it has a callback, it calls it whenever it's ready?

**Review by bernardnormier — APPROVED** [review]
It would be nice to add a doc-comment to ReadyCallback as well.

File: java/src/Ice/src/main/java/com/zeroc/IceInternal/Transceiver.java (hunk 2 above)
Comment (bentoi) [reply java/src/Ice/src/main/java/com/zeroc/IceInternal/Transceiver.java:19]: The thread pool always calls `setReadyCallback` with a non-null callback object. The transport is free to use it or not use it. For example:
- the TCP transceiver  implementation ignores it
-  the SSL and WS transceivers use it (in addition to the selectable channel)
- the BT transceiver implementation doesn't have a selectable channel, it only relies on the ready callback for read/write readiness

**Comment by bentoi** [comment]
> It would be nice to add a doc-comment to ReadyCallback as well.

I've added a comment, can you check again?

**Review by bernardnormier — APPROVED** [review]
Looks good.

File: java/src/Ice/src/main/java/com/zeroc/IceInternal/Transceiver.java [hunk 3]
```diff
@@ -5,8 +5,24 @@
...
+   */
   java.nio.channels.SelectableChannel fd();
 
+  /**
+   * Sets the transceiver ready callback. This method is called by the thread pool to provide a
+   * callback object that the transceiver can use to notify the thread pool's selector when more
+   * data is ready to be read or written by this transceiver.
+   *
+   * @param callback The ready callback provided by the thread pool's selector.
+   * @remark The transceiver implementation typically uses this callback when it buffers data read
```
Comment (InsertCreativityHere) [inline java/src/Ice/src/main/java/com/zeroc/IceInternal/Transceiver.java:23]: To my knowledge, JavaDoc doesn't support `@remark` tags.
Even though it's a less elegant name, this should probably be:
```suggestion
   * @implNote The transceiver implementation typically uses this callback when it buffers data read
```

Comment (bentoi) [reply java/src/Ice/src/main/java/com/zeroc/IceInternal/Transceiver.java]: Good point, I thought it worked. I've integrated the text to the summary instead.

**Pull request merged**